File: SendToVB/SendToVB/Form1.vb
Public Class Form1

    Private Sub btnBrowse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnBrowse.Click
        If OpenFileDialog1.ShowDialog = Windows.Forms.DialogResult.OK Then
            txtAttachment.Text = OpenFileDialog1.FileName
        End If
    End Sub

    Private Sub btnSend_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSend.Click
        Dim mapi As SendFileTo.MAPI
        Dim recipient As SendFileTo.MapiRecipDesc

        mapi = New SendFileTo.MAPI()
        recipient = New SendFileTo.MapiRecipDesc()

        If Not mapi.ResolveName(txtTo.Text, recipient) Then
            Exit Sub
        End If

        mapi.AddRecipient(recipient)

        If txtAttachment.Text <> "" Then
            mapi.AddAttachment(txtAttachment.Text)
        End If

        If mapi.SendMail(txtSubject.Text, txtBody.Text) Then
            If txtTo.Text <> "" And (txtAttachment.Text <> "" Or (txtBody.Text <> "" And txtSubject.Text <> "")) Then
                'message sent directly
                MessageBox.Show("Message sent", "Send Mail", MessageBoxButtons.OK, MessageBoxIcon.Information)
            Else
                'will open browser window to edit mail and send manually
            End If
        End If
    End Sub

End Class

File: SendToCS/SendToCS/Form1.cs
using System;
using System.Windows.Forms;

namespace Test
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void OnBrowseClicked(object sender, MouseEventArgs e)
        {
            if (DialogResult.OK == openFileDialog1.ShowDialog())
                txtAttachment.Text = openFileDialog1.FileName;
        }

        private void btnSend_Click(object sender, EventArgs e)
        {
            SendFileTo.MAPI mapi;
            SendFileTo.MapiRecipDesc recipient;

            mapi = new SendFileTo.MAPI();
            recipient = new SendFileTo.MapiRecipDesc();

            if (! mapi.ResolveName(txtTo.Text, ref recipient) )
                return;

            mapi.AddRecipient(recipient);

            if (txtAttachment.Text != "" )
                mapi.AddAttachment(txtAttachment.Text);

            if ( mapi.SendMail(txtSubject.Text, txtBody.Text) )
                if (txtTo.Text != "" && (txtAttachment.Text != "" || (txtBody.Text != "" && txtSubject.Text != "")) )
                {
                    //message sent directly
                    MessageBox.Show("Message sent", "Send Mail", MessageBoxButtons.OK, MessageBoxIcon.Information);
                } else {
                    //will open browser window to edit mail and send manually
                }
        }
    }
}